sdv/constraints/tabular.py

```diff
@@ -475,3 +476,96 @@ def reverse_transform(self, table_data):
         table_data[self._column] = self._formula(table_data)
 
         return table_data
+
+
+class OneHotEncoding(Constraint):
+    """Ensure the appropriate columns are one hot encoded.
+
+    This constraint allows the user to specify a list of columns where each row
+    is a one hot vector. During the reverse transform, the output of the model
+    is transformed so that the column with the largest value is set to 1 while
+    all other columns are set to 0.
+
+    Args:
+        columns (list[str]):
+            Names of the columns containing one hot rows.
+        handling_strategy (str):
+            How this Constraint should be handled, which can be ``transform``
+            or ``reject_sampling`` (not recommended). Defaults to ``transform``.
+    """
+
+    def __init__(self, columns, handling_strategy='transform'):
+        self._columns = columns
+        self.constraint_columns = tuple(columns)
+        super().__init__(handling_strategy, fit_columns_model=True)
+
+    def _sample_constraint_columns(self, table_data):
+        table_data = table_data.copy()
+
+        condition_columns = [col for col in self._columns if col in table_data.columns]
+        is_one = table_data[condition_columns].values == 1.0
+        is_zero = table_data[condition_columns].values == 0.0
+        if not (is_one | is_zero).all():
+            raise ValueError('Condition values must be ones or zeros.')
+
+        if (table_data[condition_columns].sum(axis=1) > 1.0).any():
+            raise ValueError('Each row of a condition can only contain one number one.')
+
+        if (table_data[condition_columns].sum(axis=1) == 0.0).any():
+            proposed_table_data = super()._sample_constraint_columns(table_data)
```

Comment:
Before sampling we should check how many columns are left. It could be the case where there is only one column left, so then we directly know that the missing value must be 1